source/common/upstream/upstream_impl.cc

```diff
@@ -397,6 +397,28 @@ HostSet& PrioritySetImpl::getOrCreateHostSet(uint32_t priority,
   return *host_sets_[priority];
 }
 
+HostSet&
+PrioritySetImpl::getOrCreateMutableHostSet(uint32_t priority,
+                                           absl::optional<uint32_t> overprovisioning_factor) {
+  getOrCreateHostSet(priority, overprovisioning_factor);
```

Comment:
In your description, you claim the goal is to make `HostSet&` always const, but this seems an exception..

Reply:
Fixed this, now we also enforce that updates to the subset host sets have to go through the priority set